src/System.Windows.Forms.Primitives/src/Interop/User32/Interop.EnumWindows.cs

```
@@ -36,7 +36,12 @@ public static BOOL EnumWindows(EnumWindowsCallback lpEnumFunc)
 
         private static BOOL HandleEnumWindowsNativeCallback(IntPtr hWnd, IntPtr lParam)
         {
-            return ((EnumWindowsCallback)GCHandle.FromIntPtr(lParam).Target)(hWnd);
+            if (GCHandle.FromIntPtr(lParam).Target is EnumWindowsCallback callback)
+            {
```

As with the other enum callbacks bang and fail fast is a better choice.